server.py

```diff
@@ -98,11 +170,16 @@ def contest(code, number):
 @app.get("/contest/<code>")
 @login_required
 def contest(code):
-    if not code in contests:
-        return "Contest does not exist"
-    if contests[code].start_time > datetime.datetime.now():
-        return "The contest had not started yet."
-    return bottle.template("contest.html", code=code, contest=contests[code])
+    if not Contest.select().where(Contest.code == code).exists():
+        return bottle.abort(404, "no such contest")
+    contest = Contest.get(Contest.code == code)
+    questions = (
```

Comment:
If you look at the [Foreign Key docs](http://docs.peewee-orm.com/en/latest/peewee/api.html#ForeignKeyField) an ORM lets you avoid writing SQL. What you're doing here is basically writing SQL in python.

We could do something like:
```python
questions = list(contest.questions`
```